test-requirements.txt

```diff
@@ -1,7 +1,9 @@
-pytest==6.1.2
+pytest==6.2.5
 pytest-forked==1.3.0
 pytest-localserver==0.5.0
 pytest-cov==2.10.1
+parsimonious==0.10.0
```

Comment:
The last time we upgraded parsimonious past 0.8.1, we noticed a big performance hit. However I think that was version 0.9.x, so maybe 0.10.0 resolved it?

I think you should do a performance test between what we use in Snuba (0.8.1) and 0.10.0. If we can start upgrading parsimonious again I'd be pretty happy.

Reply:
Parsimonious 0.8.1 is unsupported for python 3.11.

Errors with:
```
ImportError: cannot import name 'getargspec' from 'inspect' (/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/inspect.py)
```
because of deprecated `getargspec()`. Should use `inspect.signature()` or `inspect.getfullargspec()` instead.

Reply:
We use Python 3.8 in prod I believe. There's a push to get to 3.10 but it's not there yet.

This is good to know though. If the perf tests are OK we should upgrade.

Reply:
Perfect, thanks for that! Would be good to link to any results you have.